esp/src/src-react/components/forms/Fields.tsx

```diff
@@ -490,11 +490,17 @@ export const TargetGroupTextField: React.FunctionComponent<TargetGroupTextFieldP
 
     React.useEffect(() => {
         TpGroupQuery({}).then(({ TpGroupQueryResponse }) => {
-            setTargetGroups(TpGroupQueryResponse.TpGroups.TpGroup.map(n => {
-                return {
-                    key: n.Name,
-                    text: n.Name + (n.Name !== n.Kind ? ` (${n.Kind})` : "")
-                };
+            setTargetGroups(TpGroupQueryResponse.TpGroups.TpGroup.map(group => {
+                switch (group?.Kind) {
+                    case "Thor":
+                    case "hthor":
+                    case "Roxie":
+                    case "Plane":
+                        return {
+                            key: group.Name,
+                            text: group.Name + (group.Name !== group.Kind ? ` (${group.Kind})` : "")
+                        };
+                }
```

Comment:
This will map to `undefined` for any Kind not listed?  Should it have a filter after the map?